### PR TITLE
Add persist driver cache

### PR DESCRIPTION
**Pull request by marcheffels**
## ⚡ PowerGrid - Pull Request

Welcome and thank you for your interest in contributing to our project!. You must use this template to submit a Pull Request or it will not be accepted.

--- 
#### Motivation

- [ ] Bug fix
- [ ] Enhancement
- [x] New feature
- [ ] Breaking change

#### Description

This Pull Request adds persist driver cache.

If a project has many tables, the cookie header will be going to large.

#### Documentation

 This PR requires [Documentation](https://github.com/Power-Components/powergrid-doc) update?

- [X] Yes
- [] No
- [ ] I have already submitted a Documentation pull request.

### CODE REVIEW

File: src/Concerns/Persist.php
Comment (dansysanalyst) [inline src/Concerns/Persist.php:1]: I suggest: 

```php
    private function getPersistDriverStoreConfig(): string
    {
        return strval(config('livewire-powergrid.persist_driver_store'));
    }
```

and perhaps we could profit and improve code readability with:

```php
        $jsonState = strval(json_encode($state));

        match ($this->getPersistDriverConfig()) {
            'session' => Session::put($this->getPersistKeyName(), $jsonState),
            'cache'   => Cache::store($this->getPersistDriverStoreConfig())->put($this->getPersistKeyName(), $jsonState),
            default   => Cookie::queue($this->getPersistKeyName(), $jsonState, now()->addYears(5)->unix())
        };
```

**Comment by dansysanalyst** [comment]
Hi @marcheffels,

Thanks for the PR!

Could you please run `composer pint:fix` ?

**Comment by marcheffels** [comment]
Hi @dansysanalyst,

very gladly, thx

**Comment by luanfreitasdev** [comment]
Thank you @marcheffels

**Pull request merged**